hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestFileChecksum.java

```diff
@@ -575,6 +596,8 @@ private FileChecksum getFileChecksum(String filePath, int range,
       dnIdxToDie = getDataNodeToKill(filePath);
       DataNode dnToDie = cluster.getDataNodes().get(dnIdxToDie);
       shutdownDataNode(dnToDie);
+      // wait enough time for the locations to be updated.
+      Thread.sleep(STALE_INTERVAL);
```

Comment:
Should we waitFor instead?

Reply:
Yes, I agree with you @goiri .
I experimented with wait for number of live replicas: this did not work. It stayed 8 and did not go back to 9.
Do you have suggestion what conditions should we be waiting for?

Reply:
I am not very close to this part of the code but there must be ways to force the statistics to update.
Not sure who can help with this part of the code.

Reply:
I see. the problem is that I cannot reproduce it on my local machine. However, it seems that it fails in a consistent way on Yetus.
If it is not a real bug, I wonder if volumeScanner could be a factor in randomly slowing down the DNs. I see many log message from the volume scanner when I run locally.

Reply:
I am not sure exactly what test you are interested, but here you can see one of the failed tests full logs:
https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2408/2/testReport/org.apache.hadoop.hdfs/TestFileChecksum/testStripedFileChecksumWithMissedDataBlocksRangeQuery11/

Here are all the tests:
https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-2408/2/testReport/org.apache.hadoop.hdfs/

Reply:
I see there is truncation there too though.
We may want to make the test a little less verbose.

Reply:
Thanks @goiri 
Those are the logs I was looking at. All logs in `TestFileChecksum` and `TestFileChecksumCompositeCrc` truncate the last 9 seconds prior to the failure.

Reply:
I could reproduce the failure locally:
```
$ ./start-build-env.sh
$ mvn clean install -DskipTests -Pnative
$ cd hadoop-hdfs-project/hadoop-hdfs
$ mvn test -Pnative -Pparallel-tests
```
Attached the stdout in the JIRA: https://issues.apache.org/jira/secure/attachment/13014321/org.apache.hadoop.hdfs.TestFileChecksum-output.txt

Reply:
I could reproduce even without `-Pparallel-tests`
```
$ pwd
/home/aajisaka/hadoop/hadoop-hdfs-project/hadoop-hdfs
$ mvn test -Dtest=TestFileChecksum -Pnative
```